common/djangoapps/course_modes/views.py

```diff
@@ -188,7 +188,10 @@ def get(self, request, course_id, error=None):
             "error": error,
             "responsive": True,
             "nav_hidden": True,
-            "content_gating_enabled": ContentTypeGatingConfig.enabled_for_course(course_key=course_key),
+            "content_gating_enabled": ContentTypeGatingConfig.enabled_for_enrollment(
+                user=request.user,
```

Comment:
i would just verify if you haven't already that the track selection page still loads without errors if you're not enrolled, though if it displays the correct thing that would be great

Reply:
@MatthewPiatetsky 
thanks, have tested it locally both enrolled and unrolled trying to access the course track page, with and without user in holdback. Seem to get the correct/appropriate track selection page wording in all cases. Just need to fix up the broken tests.